commit b340c0b1ebd25bb788123b41a07154e088bbe5f7 (refs/changes/49/1049/10)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-06-23T19:57:29+02:00 (2 years, 4 months ago)
    
    MANTA-2999 muskie: Upgrade Node.js version from 0.10 to 4.x (or 6.x) before its EOL in Oct 2016
    Reviewed by: David Pacheco <dap@joyent.com>
    Reviewed by: Jordan Hendricks <jordan.hendricks@joyent.com>

diff --git a/Makefile b/Makefile
index b337435..ffd2f8a 100644
--- a/Makefile
+++ b/Makefile
@@ -55,8 +55,8 @@ CLEAN_FILES += node_modules
 #
 NAME 			= muskie
 NODE_PREBUILT_TAG       = zone
-NODE_PREBUILT_VERSION	:= v0.10.48
-NODE_PREBUILT_IMAGE     = fd2cc906-8938-11e3-beab-4359c665ac99
+NODE_PREBUILT_VERSION	:= v4.6.1
+NODE_PREBUILT_IMAGE     = 18b094b0-eb01-11e5-80c1-175dac7ddf02
 
 include ./tools/mk/Makefile.defs
 include ./tools/mk/Makefile.node_prebuilt.defs
diff --git a/lib/common.js b/lib/common.js
index b2d72ec..3d4c52f 100644
--- a/lib/common.js
+++ b/lib/common.js
@@ -889,13 +889,6 @@ module.exports = {
         function setup(req, res, next) {
             req.config = options;
             req.moray = options.moray();
-
-            // MANTA-331: while a trailing '/' is ok in HTTP,
-            // this messes with the consistent hashing, so
-            // ensure there isn't one
-            /* JSSTYLED */
-            req._path = req._path.replace(/\/*$/, '');
-
             req.jobCache = options.jobCache;
 
             req.log = (req.log || options.log).child({
diff --git a/lib/errors.js b/lib/errors.js
index 8e1e22d..4bc6fec 100644
--- a/lib/errors.js
+++ b/lib/errors.js
@@ -537,20 +537,20 @@ function PreSignedRequestError(msg) {
 util.inherits(PreSignedRequestError, MuskieError);
 
 
-function RequestedRangeNotSatisfiableError(req, err) {
+function RangeNotSatisfiableError(req, err) {
     if (err && err._result && err._result.headers) {
         this.headers = {
             'Content-Range': err._result.headers['content-range']
         };
     }
     MuskieError.call(this, {
-        restCode: 'RequestedRangeNotSatisfiable',
+        restCode: 'RangeNotSatisfiable',
         statusCode: 416,
         message: sprintf('%s is an invalid range',
                          req.headers['range'])
     });
 }
-util.inherits(RequestedRangeNotSatisfiableError, MuskieError);
+util.inherits(RangeNotSatisfiableError, MuskieError);
 
 
 function RootDirectoryError(method, p) {
@@ -640,7 +640,7 @@ function translateError(err, req) {
     var cause;
     cause = VError.findCauseByName(err, 'ObjectNotFoundError');
     if (cause !== null) {
-        return (new restify.ResourceNotFoundError(cause,
+        return (new ResourceNotFoundError(cause,
             '%s does not exist', req.path()));
     }
 
@@ -649,9 +649,9 @@ function translateError(err, req) {
         return (cause);
     }
 
-    cause = VError.findCauseByName(err, 'RequestedRangeNotSatisfiableError');
+    cause = VError.findCauseByName(err, 'RangeNotSatisfiableError');
     if (cause !== null) {
-        return (new RequestedRangeNotSatisfiableError(req, cause));
+        return (new RangeNotSatisfiableError(req, cause));
     }
 
     cause = VError.findCauseByName(err, 'EtagConflictError');
diff --git a/lib/obj.js b/lib/obj.js
index a5cb497..622f0d7 100644
--- a/lib/obj.js
+++ b/lib/obj.js
@@ -870,7 +870,7 @@ function streamFromSharks(req, res, next) {
                 var rh = savedErr._result.headers;
                 if (req.headers['range'] !== undefined && rh['content-range']) {
                     res.setHeader('content-range', rh['content-range']);
-                    next(new restify.RequestedRangeNotSatisfiableError());
+                    next(new RangeNotSatisfiableError(req));
                     return;
                 }
             }
diff --git a/lib/picker.js b/lib/picker.js
index b2e797c..b56bdf1 100644
--- a/lib/picker.js
+++ b/lib/picker.js
@@ -407,7 +407,6 @@ Picker.prototype.choose = function choose(opts, cb) {
         }
     });
     dcs = shuffle(dcs);
-
     if ((replicas > 1 && this.multiDC && dcs.length < 2) ||
         !dcs.length ||
         (replicas > 1 && dcs.some(function (dc) {
diff --git a/lib/server.js b/lib/server.js
index e2056e4..821a9c9 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -45,7 +45,7 @@ var JOBS_LIVE_RE = /(state|status|name)=/i;
 ///--- Helpers
 
 // Always force JSON
-function formatJSON(req, res, body) {
+function formatJSON(req, res, body, callback) {
     if (body instanceof Error) {
         body = translateError(body, req);
         res.statusCode = body.statusCode || 500;
@@ -77,7 +77,7 @@ function formatJSON(req, res, body) {
     res.setHeader('Content-MD5', md5);
     res.setHeader('Content-Type', 'application/json');
 
-    return (data);
+    callback(null, data);
 }
 
 
@@ -167,7 +167,7 @@ function createServer(options, clearProxy) {
     server.pre(function routeLiveJobs(req, res, next) {
         var tmp = JOBS_ROOT_RE.exec(req.path());
         if (tmp && tmp.length > 1 && JOBS_LIVE_RE.test(req.query()))
-            req._path = '/' + tmp[1] + '/jobs/live';
+            req.url = url.parse('/' + tmp[1] + '/jobs/live');
 
         next();
     });
@@ -177,7 +177,7 @@ function createServer(options, clearProxy) {
 
     server.use(common.earlySetupHandler(options));
     server.use(restify.dateParser(options.maxRequestAge || 300));
-    server.use(restify.queryParser());
+    server.use(restify.queryParser({allowDots: false, plainObjects: false}));
     server.use(common.authorizationParser);
     server.use(auth.checkIfPresigned);
     server.use(common.enforceSSLHandler(options));
@@ -468,7 +468,8 @@ function addMultipartUploadRoutes(server) {
     }, uploads.createHandler());
 
     server.put({
-        path: '/:account/uploads'
+        path: '/:account/uploads',
+        contentType: 'application/json'
     }, forbiddenHandler);
 
     // Redirects
@@ -560,11 +561,13 @@ function addMultipartUploadRoutes(server) {
     }, forbiddenHandler);
 
     server.put({
-        path: '/:account/uploads/[0-f]+/:id/state'
+        path: '/:account/uploads/[0-f]+/:id/state',
+        contentType: 'application/json'
     }, forbiddenHandler);
 
     server.post({
-        path: '/:account/uploads/[0-f]+/:id/state'
+        path: '/:account/uploads/[0-f]+/:id/state',
+        contentType: 'application/json'
     }, forbiddenHandler);
 
     server.del({
@@ -579,7 +582,8 @@ function addMultipartUploadRoutes(server) {
      */
     server.post({
         path: '/:account/uploads/[0-f]+/:id/abort',
-        name: 'AbortUpload'
+        name: 'AbortUpload',
+        contentType: 'application/json'
     }, uploads.abortHandler());
 
     server.get({
@@ -587,7 +591,8 @@ function addMultipartUploadRoutes(server) {
     }, forbiddenHandler);
 
     server.put({
-        path: '/:account/uploads/[0-f]+/:id/abort'
+        path: '/:account/uploads/[0-f]+/:id/abort',
+        contentType: 'application/json'
     }, forbiddenHandler);
 
     server.head({
@@ -615,7 +620,8 @@ function addMultipartUploadRoutes(server) {
     }, forbiddenHandler);
 
     server.put({
-        path: '/:account/uploads/[0-f]+/:id/commit'
+        path: '/:account/uploads/[0-f]+/:id/commit',
+        contentType: 'application/json'
     }, forbiddenHandler);
 
     server.head({
@@ -650,7 +656,8 @@ function addMultipartUploadDataPlaneRoutes(server) {
     }, forbiddenHandler);
 
     server.post({
-        path: '/:account/uploads/[0-f]+/:id/:partNum'
+        path: '/:account/uploads/[0-f]+/:id/:partNum',
+        contentType: '*/*'
     }, forbiddenHandler);
 
     server.del({
diff --git a/lib/shark_client.js b/lib/shark_client.js
index a4dc1d8..da6568f 100644
--- a/lib/shark_client.js
+++ b/lib/shark_client.js
@@ -134,13 +134,19 @@ function request(thisp, method, opts, cb) {
     cb = once(cb);
 
     var log = thisp.log;
+
+    var headers = {
+        connection: 'keep-alive',
+        'x-request-id': opts.requestId
+    };
+
+    if (opts.range !== undefined) {
+        headers.range = opts.range;
+    }
+
     var _opts = {
         connectTimeout: thisp.connectTimeout,
-        headers: opts.headers || {
-            connection: 'keep-alive',
-            'x-request-id': opts.requestId,
-            range: opts.range
-        },
+        headers: opts.headers || headers,
         hostname: thisp.hostname,
         method: method,
         port: thisp.port
diff --git a/package.json b/package.json
index 4b3a25d..13e217a 100644
--- a/package.json
+++ b/package.json
@@ -6,42 +6,42 @@
     "private": true,
     "repository": {
         "type": "git",
-        "url": "git+ssh://git@github.com:joyent/muskie.git"
+        "url": "https://github.com/joyent/muskie.git"
     },
     "dependencies": {
-        "aperture-config": "git+ssh://git@github.com:joyent/aperture-config.git#master",
-        "assert-plus": "0.1.5",
+        "aperture-config": "https://github.com/joyent/aperture-config.git#master",
+        "assert-plus": "1.0.0",
         "backoff": "2.3.0",
-        "bunyan": "0.22.1",
-        "bunyan-syslog": "0.2.2",
+        "bunyan": "1.8.1",
+        "bunyan-syslog": "0.3.1",
         "cueball": "2.2.3",
         "dashdash": "1.3.2",
         "deep-equal": "0.0.0",
-        "dtrace-provider": "0.2.8",
-        "http-signature": "1.1.0",
+        "dtrace-provider": "0.8.0",
+        "http-signature": "1.1.1",
         "jsprim": "^1.3.1",
         "kang": "1.1.0",
         "keep-alive-agent": "0.0.1",
-        "keyapi": "git+ssh://git@github.com:joyent/keyapi.git#c30dd2710ad2175095dc0e96479686fa774b8063",
-        "libmanta": "git+ssh://git@github.com:joyent/node-libmanta.git#master",
-        "libuuid": "0.1.2",
+        "keyapi": "https://github.com/joyent/keyapi.git#e14b3d582e1d9d338b7082d61f34ba8d1bbc540a",
+        "libmanta": "https://github.com/joyent/node-libmanta.git#MANTA-3031",
+        "libuuid": "0.2.1",
         "lru-cache": "2.3.1",
         "lstream": "0.0.4",
         "mahi": "2.0.1",
-        "marlin": "git+ssh://git@github.com:joyent/manta-marlin.git#master",
+        "marlin": "https://github.com/joyent/manta-marlin.git#dev-MANTA-3039",
         "mime": "1.2.11",
         "moray": "3.1.1",
         "once": "1.3.0",
-        "restify": "2.6.3",
+        "restify": "4.3.0",
         "vasync": "1.4.3",
         "verror": "^1.9.0",
         "watershed": "0.3.0",
         "xtend": "2.1.1"
     },
     "devDependencies": {
-        "smartdc": "git+ssh://git@github.com:joyent/node-smartdc.git#v7.3",
+        "smartdc": "8.1.0",
         "manta": "4.3.0",
-        "nodeunit": "0.9.1",
+        "nodeunit": "0.10.2",
         "node-uuid": "1.4.1"
     },
     "scripts": {
diff --git a/test/mpu/abort.test.js b/test/mpu/abort.test.js
index 41a0ccc..766b22d 100644
--- a/test/mpu/abort.test.js
+++ b/test/mpu/abort.test.js
@@ -87,13 +87,13 @@ test('abort upload: upload already aborted', function (t) {
             }
 
             self.abortUpload(self.uploadId, function (err3) {
-                if (ifErr(t, err, 'aborted upload')) {
+                if (ifErr(t, err3, 'aborted upload')) {
                     t.end();
                     return;
                 }
 
                 self.getUpload(self.uploadId, function (err4, upload) {
-                    if (ifErr(t, err, 'got upload')) {
+                    if (ifErr(t, err4, 'got upload')) {
                         t.end();
                         return;
                     }
@@ -129,10 +129,11 @@ test('abort upload: upload already committed', function (t) {
             self.abortUpload(self.uploadId, function (err3) {
                 t.ok(err3);
                 if (!err3) {
-                    return (t.end());
+                    t.end();
+                    return;
                 }
                 t.ok(verror.hasCauseWithName(err3,
-                    'InvalidMultipartUploadStateError'), err);
+                    'InvalidMultipartUploadStateError'), err3);
                 t.end();
             });
         });
diff --git a/test/obj.test.js b/test/obj.test.js
index c4863a1..104c6f9 100644
--- a/test/obj.test.js
+++ b/test/obj.test.js
@@ -597,6 +597,15 @@ test('put parent not directory', function (t) {
 
 
 test('put too big', function (t) {
+    // Since MANTA-2510 we use multiDC=false and ignoreSize=true config for
+    // COAL. Therefore, this test will never throw a NotEnoughSpaceError but
+    // instead an UploadTimeoutError. Avoid running it if we find we're
+    // running in COAL:
+    if (process.env.SDC_URL && process.env.SDC_URL.indexOf('coal') !== -1) {
+        t.end();
+        return;
+    }
+
     var opts = {
         size: 1000000000000000,
         type: 'text/plain'
diff --git a/tools/jsl.node.conf b/tools/jsl.node.conf
index 76a1fc1..3d238c5 100644
--- a/tools/jsl.node.conf
+++ b/tools/jsl.node.conf
@@ -179,6 +179,7 @@
 +define NotImplementedError
 +define ParentNotDirectoryError
 +define PreSignedRequestError
++define RangeNotSatisfiableError
 +define RequestEntityTooLargeError
 +define ResourceNotFoundError
 +define RootDirectoryError
